code/AssetLib/glTF2/glTF2Asset.inl

```diff
@@ -1628,6 +1628,22 @@ inline void Node::Read(Value &obj, Asset &r) {
                 }
             }
         }
+
+        // Read the the nodes specified under MSFT_lod extension ids and save as children node
+        if (Value *msftLodExt = FindObject(*curExtensions, "MSFT_lod")) {
+            if (Value *lodIds = FindArray(*msftLodExt, "ids")) {
```

Comment:
Following the assimp code, it looks like the `FindObject` and `FindArray` functions both throw an `DeadlyImportError` exception if the `Value` object _does_ have a member with the expected name, but of a different type than the one expected. E.g. see the code for `FindObject` [here](https://github.com/Esri/assimp/blob/4859e3971f9a69e87f05cb3e8a06112f4e1bac6d/code/AssetLib/glTF/glTFCommon.h#L488-L490):

```
    if (!it->value.IsObject()) {
        throwUnexpectedTypeError("object", memberId, context, extraContext);
    }
```

So, if the `MSFT_lods` extension is misconfigured, we throw an exception where previously we didn't.

Is there a way to check the type of the value before calling one of the `Find` functions? Or can we wrap in a try/catch for that specific exception type?

Reply:
I did think about that. However this is true for any property on assimp. e.g. 
https://github.com/Esri/assimp/blob/4859e3971f9a69e87f05cb3e8a06112f4e1bac6d/code/AssetLib/glTF/glTFCommon.h#L438

or 

https://github.com/Esri/assimp/blob/4859e3971f9a69e87f05cb3e8a06112f4e1bac6d/code/AssetLib/glTF/glTFCommon.h#L424

We cannot get the type of member upfront without manually parsing every property from the document. 
glTFCommon.h implements a similar method [FindObject](https://github.com/Esri/assimp/blob/4859e3971f9a69e87f05cb3e8a06112f4e1bac6d/code/AssetLib/glTF/glTFCommon.h#L534) and [FindArray](https://github.com/Esri/assimp/blob/4859e3971f9a69e87f05cb3e8a06112f4e1bac6d/code/AssetLib/glTF/glTFCommon.h#L517). These implementations dont check the type of top level object but checks type of member as 

```
inline Value *FindObject(Value &val, const char *id) {
    Value::MemberIterator it = val.FindMember(id);
    return (it != val.MemberEnd() && it->value.IsObject()) ? &it->value : 0;
}
```

I didnt want to follow a different pattern from what is being used.

Reply:
Thanks for the examples - looks like that's the pattern Assimp uses.

If it's OK for misconfigured LOD metadata to cause the model load to fail, then nothing more to do here.